File: app/src/main/java/com/example/gpstracker/MainActivity.java
package com.example.gpstracker;

import android.Manifest;
import android.app.Activity;
import android.content.pm.PackageManager;
import android.location.Location;
import android.location.LocationListener;
import android.location.LocationManager;
import android.os.Bundle;
import android.widget.TextView;
import androidx.core.app.ActivityCompat;

public class MainActivity extends Activity implements LocationListener {

    TextView locationText;
    LocationManager locationManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        locationText = new TextView(this);
        setContentView(locationText);

        locationManager = (LocationManager) getSystemService(LOCATION_SERVICE);
        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, 1);
            return;
        }
        locationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER, 5000, 0, this);
    }

    @Override
    public void onLocationChanged(Location location) {
        locationText.setText("Lat: " + location.getLatitude() + "\nLng: " + location.getLongitude());
    }
}

File: app/src/main/java/com/example/gpstracker/MainActivity.kt
package com.example.gpstracker

import android.app.Activity
import android.os.Bundle

class MainActivity : Activity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        
        val textView = android.widget.TextView(this)
        textView.text = "GPS Tracker - Ready!"
        textView.textSize = 20f
        textView.gravity = android.view.Gravity.CENTER
        
        setContentView(textView)
    }
}
